### PR TITLE
Page Object for Settings

### PR DESCRIPTION
**Pull request by MaksymNanyzniuk**
SettingsPage class created. SettingsPageTests refactored according to Page Object pattern.

### CODE REVIEW

File: web.test.tests/web.test.tests/Pages/SettingsPage.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        public SelectElement SelectNumberFormat => new SelectElement(_driver.FindElement(By.XPath(XPathNumberFormatSelect)));
+        public SelectElement SelectDefaultCurrency => new SelectElement(_driver.FindElement(By.XPath(XPathCurrencySelect)));
+
+        public void ClickSave()
+        {
+            _driver.FindElement(By.Id("save")).Click();
+            new WebDriverWait(_driver, TimeSpan.FromMilliseconds(10000)).Until(ExpectedConditions.AlertIsPresent());
+            IAlert alert = _driver.SwitchTo().Alert();
+            Assert.AreEqual("Changes are saved!", alert.Text);
+            alert.Accept();
```
Comment (varajan) [inline web.test.tests/web.test.tests/Pages/SettingsPage.cs]: Don't do asserts in page object

File: web.test.tests/web.test.tests/Pages/SettingsPage.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        public IWebElement CancelBtn => _driver.FindElement(By.Id("cancel"));
+        public IWebElement Logout => _driver.FindElement(By.XPath("//div[text()='Logout']"));
+
+        public SelectElement SelectDateFormat => new SelectElement(_driver.FindElement(By.XPath(XPathDateFormatSelect)));
+        public SelectElement SelectNumberFormat => new SelectElement(_driver.FindElement(By.XPath(XPathNumberFormatSelect)));
+        public SelectElement SelectDefaultCurrency => new SelectElement(_driver.FindElement(By.XPath(XPathCurrencySelect)));
+
+        public void ClickSave()
+        {
```
Comment (varajan) [inline web.test.tests/web.test.tests/Pages/SettingsPage.cs]:
```suggestion
        public void Save()
        {
```

File: web.test.tests/web.test.tests/Tests/DepositePageTests.cs [hunk 3]
```diff
@@ -145,17 +151,24 @@ public void Income_And_Interest()
         [TestCase(229, fin_year_360)]
         [TestCase(360, fin_year_360)]
         [TestCase(365, fin_year_365)]
-        public void End_Date(int term, String fin_year)
+        public void End_Date(int term, String fin_year) //Question:Result value appearing?
```
Comment (varajan) [inline web.test.tests/web.test.tests/Tests/DepositePageTests.cs]: remove coment

File: web.test.tests/web.test.tests/Tests/DepositePageTests.cs [hunk 4]
```diff
@@ -166,21 +179,16 @@ public void End_Date(int term, String fin_year)
...
             IWebElement term_field = driver.FindElement(By.Id("term"));
-
-            String exp_end_date = start_date.AddDays(term).ToString("dd/MM/yyyy", CultureInfo.InvariantCulture);
-
             term_field.Clear();
             term_field.SendKeys(term.ToString());
 
             String act_end_date = driver.FindElement(By.Id("endDate")).GetAttribute("value");
 
+            Console.WriteLine($"ER: {exp_end_date}; AR: {act_end_date}");
```
Comment (varajan) [inline web.test.tests/web.test.tests/Tests/DepositePageTests.cs]: Don't need to write to Console

**Review by varajan — COMMENTED** [review]
Looks good.

File: web.test.tests/web.test.tests/Pages/SettingsPage.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            IAlert alert = _driver.SwitchTo().Alert();
+            Assert.AreEqual("Changes are saved!", alert.Text);
+            alert.Accept();
+            new WebDriverWait(_driver, TimeSpan.FromMilliseconds(10000)).Until(ExpectedConditions.VisibilityOfAllElementsLocatedBy(By.Id("amount")));
+        }
+
+        public String GetSelectedDateFormatText()
+        {
+            return _driver.FindElement(By.XPath(XPathDateFormatSelect)).GetAttribute("value");
+        }
```
Comment (varajan) [inline web.test.tests/web.test.tests/Pages/SettingsPage.cs]:
```suggestion
public string SelectedDateFormat => SelectDateFormat.SelectedOption.Text;
```

File: web.test.tests/web.test.tests/Pages/SettingsPage.cs (hunk 1 above)
Comment (MaksymNanyzniuk) [reply web.test.tests/web.test.tests/Pages/SettingsPage.cs]: if (alert.Text == "Changes are saved!") alert.Accept();

File: web.test.tests/web.test.tests/Tests/DepositePageTests.cs (hunk 3 above)
Comment (MaksymNanyzniuk) [reply web.test.tests/web.test.tests/Tests/DepositePageTests.cs]: Removed

**Pull request merged**